foundationdb/init-foundationdb.sh

```diff
@@ -0,0 +1,171 @@
+        shift
+        continue
+    fi
+    if [ "${1}" = "--fdbcli-command" ]; then
+        shift
+        if [ -z "${1}" ]; then
+            usage
+        fi
+        FDBCLI_COMMAND="${1}"
+        shift
+        continue
+    fi
+    usage
+done
+
+if [ -z "${CLUSTER_FILE_CONTENT}" ] || [ -z "${HOST_TO_LISTEN_ON}" ]; then
+    usage
+fi
+
+CLUSTER_SERVERS=$(echo "${CLUSTER_FILE_CONTENT}" | awk -F '@' '{ print $2 }' | tr '\r\n' ' ' | sed 's/,/ /g' )
+HOST_SERVER_IDS=
+for server_address in ${CLUSTER_SERVERS} ; do
+  host=$(echo "${server_address}" | awk -F ':' '{ print $1 }')
+  port=$(echo "${server_address}" | awk -F ':' '{ print $2 }')
+  if [ "${host}" = "${HOST_TO_LISTEN_ON}" ] ; then
+    HOST_SERVER_IDS+="${port} "
+  fi
+done
+
+if [ -z "${HOST_SERVER_IDS}" ] ; then
+  log "Cluster file doesn't declare any servers for a current host (${HOST_TO_LISTEN_ON})"
+  log 'No FDB servers will be started on the current host' 'WARNING'
+fi
+
+
+#####################################
+### Actual installation procedure ###
+#####################################
+
+
+### Installing FDB clients package ###
+#------------------------------------#
+${SCRIPT_DIR}/download_artifacts.sh
+FDB_CLIENTS_PKG='foundationdb-clients_6.1.8-1_amd64.deb'
+FDB_SERVER_PKG='foundationdb-server_6.1.8-1_amd64.deb'
+
+log "Installing ${FDB_CLIENTS_PKG}"
+dpkg --install ${PACKAGE_CACHE}/foundationdb-clients_6.1.8-1_amd64.deb
+
+log "Installing ${FDB_SERVER_PKG}"
+dpkg --install ${PACKAGE_CACHE}/foundationdb-server_6.1.8-1_amd64.deb
+
+
+### Ensuring FDB directories are accessible ###
+#---------------------------------------------#
+log 'Making sure FDB directories are created and are owned by foundationdb user.'
+mkdir -pv /var/log/foundationdb
+chown -R foundationdb:foundationdb /var/log/foundationdb
+mkdir -pv "${DATA_DIR}"
+chown -R foundationdb:foundationdb "${DATA_DIR}"
+
+
+### Filling /etc/foundationdb/fdb.cluster ###
+#-------------------------------------------#
+CLUSTER_FILE=/etc/foundationdb/fdb.cluster
+log "Filling ${CLUSTER_FILE} file"
+if [ -f "${CLUSTER_FILE}" ]; then
+  cp ${CLUSTER_FILE} "${CLUSTER_FILE}.$(date +'%Y-%m-%d_%H-%M-%S')"
+fi
+echo "${CLUSTER_FILE_CONTENT}" > "${CLUSTER_FILE}"
+
+
+### Filling /etc/foundationdb/foundationdb.conf ###
+#-------------------------------------------------#
+CONF_FILE=/etc/foundationdb/foundationdb.conf
+log "Filling ${CONF_FILE} file"
+if [ -f "${CONF_FILE}" ]; then
+  cp ${CONF_FILE} "${CONF_FILE}.$(date +'%Y-%m-%d_%H-%M-%S')"
+fi
+export HOST_TO_LISTEN_ON
+export DATA_DIR
+export FDB_SERVERS=$(
+  for server_id in ${HOST_SERVER_IDS}; do
+    echo "[fdbserver.${server_id}]"
+  done
+)
+envsubst '$HOST_TO_LISTEN_ON $DATA_DIR $FDB_SERVERS'\
+ < "${SCRIPT_DIR}/foundationdb.conf" > "${CONF_FILE}"
+
+
+systemctl enable foundationdb.service
+systemctl start foundationdb.service
+systemctl status foundationdb.service
+
+
+log 'Waiting for FDB cluster to start'
+wait_start=$(date +%s)
+while ! fdbcli --exec status
```

Comment:
I think this will return a successful exit code even if the database is unavailable.